kmir/src/tests/integration/test_integration.py

```diff
@@ -335,39 +333,11 @@ def test_exec_smir(
 )
 def test_prove_termination(test_data: tuple[str, Path], tmp_path: Path, kmir: KMIR) -> None:
     testname, smir_json = test_data
-    spec_file = tmp_path / f'{testname}.k'
-    gen_opts = GenSpecOpts(smir_json, spec_file, 'main')
-
-    proof_dir = tmp_path / 'proof'
-    prove_opts = ProveRawOpts(spec_file, proof_dir=proof_dir)
-
-    _kmir_gen_spec(gen_opts)
-    _kmir_prove_raw(prove_opts)
-
-    claim_labels = kmir.get_claim_index(spec_file).labels()
-    for label in claim_labels:
-        proof = Proof.read_proof_data(proof_dir, label)
-        assert proof.passed
 
+    prove_rs_opts = ProveRSOpts(rs_file=smir_json, smir=True)
 
-PROVING_DIR = (Path(__file__).parent / 'data' / 'proving').resolve(strict=True)
-PROVING_FILES = list(PROVING_DIR.glob('*-spec.k'))
-
-
-@pytest.mark.parametrize(
-    'spec',
-    PROVING_FILES,
-    ids=[spec.stem for spec in PROVING_FILES],
-)
-def test_prove(spec: Path, tmp_path: Path, kmir: KMIR) -> None:
-    proof_dir = tmp_path / (spec.stem + 'proofs')
-    prove_opts = ProveRawOpts(spec, proof_dir=proof_dir)
-    _kmir_prove_raw(prove_opts)
-
-    claim_labels = kmir.get_claim_index(spec).labels()
-    for label in claim_labels:
-        proof = Proof.read_proof_data(proof_dir, label)
-        assert proof.passed
+    proof = KMIR.prove_rs(prove_rs_opts)
+    assert proof.passed
```

Comment:
I have removed the single test that we had which was based on `gen-spec` and `prove-raw`, as well as switching the `test_prove_termination` suite to using `prove_rs`. This means the `gen-spec` and `prove-raw` commands are now untested. I would suggest removing them because they don't fit well with the changed execution in this PR.